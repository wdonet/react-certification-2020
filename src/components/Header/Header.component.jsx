import React from 'react';
import { useHistory } from 'react-router';
import Switch from '../Switch';
import { HeaderContainer, Loggin, Menu, SearchInput, SearchLabel } from './Header.styled';
import { DARK_THEME } from '../../utils/constants';

function Header() {
  const history = useHistory();

  function search({ key, target }) {
    if (key === 'Enter') {
      history.push(`/search/${target.value}`);
    }
  }

  return (
    <HeaderContainer>
      <Menu
        blurImage="/images/menu.svg"
        focusImage="/images/menu-focus.svg"
        clickHandler={() => console.log('Menu')}
        alt="Menu"
        name="Menu"
      />
      <SearchLabel htmlFor="search-input">search</SearchLabel>
      <SearchInput
        id="search-input"
        placeholder="Search for videos..."
        onKeyDown={search}
      />
      <Switch
        valueName={DARK_THEME}
        textOn="☽"
        textOff="☼"
        onToggle={(darkTheme) => console.log(darkTheme ? 'Dark' : 'Light')}
      />
      <Loggin
        blurImage="/images/loggin.svg"
        focusImage="/images/loggin-focus.svg"
        clickHandler={() => console.log('Loggin')}
        alt="Loggin"
        name="Loggin"
      />
    </HeaderContainer>
  );
}

export default Header;
