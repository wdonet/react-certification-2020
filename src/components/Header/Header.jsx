import React from 'react';
import { Navbar, Form, Col } from 'react-bootstrap';
import { HiMenu } from 'react-icons/hi';

import { Button } from '../Buttons/Button';
import InputText from '../TextInput/TextInput';
import ProfileImage from '../ProfileImage/ProfileImage';

function Header() {
  return (
    <Navbar bg="light" expand="lg">
      <Col sm={6}>
        <Button>
          <HiMenu />
        </Button>
        <InputText placeholder="Search" />
      </Col>
      <Col sm={{ offset: 3, span: 3 }}>
        <Form inline>
          <Form.Check type="switch" id="custom-switch" label="Dark mode" />
          <ProfileImage src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s88-c-k-c0x00ffffff-no-rj" />
        </Form>
      </Col>
    </Navbar>
  );
}

export default Header;
