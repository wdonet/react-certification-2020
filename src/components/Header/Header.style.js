import styled from "styled-components";


const header = styled.div`
    width: 100%;
    background-color: whitesmoke;
    opacity: 80%;
`

const burgerMenu = styled.div`
    display: inline-block;
    width: 80px;
`

const inputWrapper = styled.div`
    width: 70%;
    display: inline-block;
`

const inputSearch = styled.input`
    width: 250px;
    margin-left: 5%;
    height: 30px;
    border-radius: 8px;
    background: gray;
    opacity: 80%;
    border: none;
    color: white;
    padding: 0px 20px;
    &::-webkit-input-placeholder { 
        color: white; 
        font-size: 15px; 
    } 
`

const toggleWrapper = styled.div`
    position: relative;
    /* justify-content: left; */
    width: 15%;
    display: inline-block;
    padding-left: 3%
`

const toggleBox = styled.label`
    position: absolute;
    top: 13px;
    /* left: 0; */
    width: 42px;
    height: 26px;
    border-radius: 15px;
    background: #bebebe;
    cursor: pointer;
    &::after {
        content: "";
        display: block;
        border-radius: 50%;
        width: 18px;
        height: 18px;
        margin: 3px;
        background: #ffffff;
        box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.2);
        transition: 0.2s;
    }
`

const togglebutton = styled.input`
    opacity: 0;
    z-index: 1;
    border-radius: 15px;
    width: 42px;
    height: 26px;
    &:checked + ${toggleBox} {
        background: #4fbe79;
        &::after {
            content: "";
            display: block;
            border-radius: 50%;
            width: 18px;
            height: 18px;
            margin-left: 21px;
            transition: 0.2s;
        }
    }
`

const darkMode = styled.p`
    display: inline-block;
    font-size: 15px;
    margin-left: 10px;
`
const profile = styled.img`
    width: 50px;
    border-radius: 50%;
    box-shadow: 2px 2px 25px 3px black;
`


const Style = {burgerMenu, header, inputSearch, inputWrapper, toggleWrapper, toggleBox, togglebutton, darkMode, profile}
export default Style

