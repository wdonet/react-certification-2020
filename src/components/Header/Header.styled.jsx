import styled from 'styled-components';
import ImageButton from '../ImageButton';

export const HeaderContainer = styled.header`
  background-color: ${({ theme }) => theme.headerBackground};
  height: 52px;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: auto;
  font-size: 0.9rem;
`;

export const Menu = styled(ImageButton)`
  font-size: inherit;
  margin: 0 0.75em;
  width: 18px;
  flex-grow: 0;
  flex-shrink: 0;
  outline: none;
`;

export const SearchLabel = styled.label`
  display: none;
`;

export const SearchInput = styled.input`
  font-size: inherit;
  font-weight: bold;
  outline: none;
  width: 75ch;
  background-color: transparent;
  border: none;
  border-right: 1px solid rgba(255, 255, 255, 0.4);
  color: ${({ theme }) => theme.headerTextColor};
  margin-right: 0.75em;
`;

export const Loggin = styled(ImageButton)`
  font-size: inherit;
  margin: 0 0.75em;
  width: 30px;
  flex-grow: 0;
  flex-shrink: 0;
  outline: none;
`;
