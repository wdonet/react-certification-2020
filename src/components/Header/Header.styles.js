import styled from 'styled-components';

const Container = styled.div`
  width: 100%;
  vertical-align: top;
  background-color: rgba(230, 90, 232, 0.2);
  margin: 0px;
  overflow: hidden;
`;

const Menu = styled.a`
  display: inline-block;
  text-align: center;
  padding: 10px;
  float: left;
  background-color: rgba(62, 64, 163);

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }
`;

const SearchContainer = styled.div`
  background-color: rgba(62, 64, 163);
  display: inline-block;
  padding: 5px;
  margin-left: 2px;
`;

const SearchBar = styled.input`
  padding: 5px;
  display: inline-block;
  background-color: rgba(42, 239, 245);
  border: none;
  margin-top: 8px;
  margin-right: 16px;
  font-size: 17px;
  color: white;
`;

const SearchButton = styled.a`
  display: inline-block;
  padding-top: 5px;
  padding-right: 5px;
  text-align: center;
  float: right;

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }
`;

const LoginButton = styled.a`
  display: inline-block;
  text-align: center;
  padding: 10px;
  float: right;
  background-color: #3e40a3;

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }
`;

const Styled = { Container, Menu, SearchBar, LoginButton, SearchContainer, SearchButton };
export default Styled;
