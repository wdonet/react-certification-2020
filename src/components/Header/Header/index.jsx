import React from 'react';
import Styled from './styled';
import './header.css';

import Logo from '../Logo';
import SearchBar from '../SearchBar';
import Login from '../Login';

const Header = () => {
  return (
    <Styled.Header>
      <Styled.LogoImage>
        <Logo />
      </Styled.LogoImage>
      <Styled.SearchArea>
        <SearchBar />
      </Styled.SearchArea>
      <Styled.Login>
        <Login />
      </Styled.Login>
    </Styled.Header>
  );
};

export default Header;
