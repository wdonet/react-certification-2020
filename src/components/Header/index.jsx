import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { faHome, faHeart, faUser, faSignOutAlt } from '@fortawesome/free-solid-svg-icons';

import {
  HeaderContainer,
  LeftItems,
  CenterItems,
  RightItems,
  ToggleContainer,
} from './styled';

import SearchInput from '../SearchInput';
import HeaderButton from '../HeaderButton';
import Icon from '../Icon';
import Toggle from '../Toggle';
import Login from '../Login';

import { useAuth } from '../../providers/Auth';

const Header = ({ handleThemeSwitch }) => {
  const [displayLogin, setDisplayLogin] = useState(false);

  const { authenticated, logout } = useAuth();

  const showLoginModal = () => setDisplayLogin(true);
  const closeLoginModal = () => setDisplayLogin(false);

  return (
    <>
      <HeaderContainer>
        <LeftItems>
          <Link
            style={{
              textDecoration: 'none',
            }}
            to="/"
          >
            <HeaderButton>
              <Icon icon={faHome} size="xsmall" />
            </HeaderButton>
          </Link>
          <ToggleContainer>
            <Toggle onChange={handleThemeSwitch} />
          </ToggleContainer>
        </LeftItems>
        <CenterItems>
          <SearchInput />
        </CenterItems>
        <RightItems>
          {authenticated && (
            <Link
              style={{
                textDecoration: 'none',
              }}
              to="/favorites"
            >
              <HeaderButton>
                <Icon icon={faHeart} size="xsmall" />
              </HeaderButton>
            </Link>
          )}

          <HeaderButton onClick={authenticated ? logout : showLoginModal}>
            <Icon icon={authenticated ? faSignOutAlt : faUser} size="xsmall" />
          </HeaderButton>
        </RightItems>
      </HeaderContainer>
      <Login open={displayLogin} closeModal={closeLoginModal} />
    </>
  );
};

export default Header;
