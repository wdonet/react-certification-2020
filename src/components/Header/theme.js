const base = {
  easeOutBack: 'cubic-bezier(0.34, 1.56, 0.64, 1)',
  colorWhite: 'rgb(255, 255, 255)',
  colorBlack: 'rgb(0, 0, 0)',
};

const dark = {
  id: 'dark',
  ...base,
  backgroundColor: '#333',
  textColor: 'black',
};

const light = {
  id: 'light',
  ...base,
  backgroundColor: '#666',
  textColor: 'white',
};

export const theme = { dark, light };
