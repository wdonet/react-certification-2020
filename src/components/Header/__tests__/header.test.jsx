import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';

import Header from '../index';
import AuthProvider from '../../../providers/Auth';
import YoutubeDataProvider from '../../../providers/YoutubeData';

import { lightTheme } from '../../../theme';

describe('Header', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <AuthProvider>
            <YoutubeDataProvider>
              <ThemeProvider theme={lightTheme}>
                <Header />
              </ThemeProvider>
            </YoutubeDataProvider>
          </AuthProvider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
