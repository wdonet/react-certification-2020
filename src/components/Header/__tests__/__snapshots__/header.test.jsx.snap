// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Array [
  .c0 {
  background-color: #FFFFFF;
  padding: 10px 0 10px 0;
  box-shadow: 0px 2px 4px -1px rgb(0 0 0 / 20%),0px 4px 5px 0px rgb(0 0 0 / 14%), 0px 1px 10px 0px rgb(0 0 0 / 12%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  text-align: left;
}

.c9 {
  text-align: center;
  width: 40%;
}

.c13 {
  text-align: right;
}

.c3 {
  height: 100%;
}

.c10 {
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 400;
  font-style: normal;
  color: #333333;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #f0dddd85;
  border-radius: 0;
  width: 100%;
  font-size: 14px;
  line-height: 27px;
  background-color: #ECECEC;
  box-sizing: border-box;
  width: 100%;
}

.c11 {
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 0 10px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 45px;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
}

.c11:focus {
  outline: 0;
}

.c12 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 > * {
  margin-right: 10px;
}

.c2 {
  display: inline;
  padding: 0 20px 0 20px;
  cursor: pointer;
  color: #333333;
  line-height: initial;
}

.c2:hover {
  color: #eea2a2;
}

.c8 {
  border-radius: 34px;
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: #333333;
  -webkit-transition: 0.4s;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c8:before {
  border-radius: 50%;
  position: absolute;
  content: '';
  height: 13px;
  width: 13px;
  left: 2px;
  bottom: 1.5px;
  background-color: white;
  -webkit-transition: 0.4s;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c6:before {
  position: absolute;
  content: '';
  height: 13px;
  width: 13px;
  left: 2px;
  bottom: 1.5px;
  background-color: white;
  -webkit-transition: 0.4s;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c6:checked + .c7 {
  background-color: #eea2a2;
}

.c6:checked + .c7:before {
  -webkit-transform: translateX(13px);
  -ms-transform: translateX(13px);
  -webkit-transform: translateX(13px);
  -ms-transform: translateX(13px);
  transform: translateX(13px);
}

.c6:focus + .c7 {
  box-shadow: 0 0 1px #eea2a2;
}

.c4 {
  position: relative;
  display: inline-block;
  width: 30px;
  height: 16px;
}

.c4 .c5 {
  opacity: 0;
  width: 0;
  height: 0;
}

@media (max-width:900px) {
  .c0 {
    height: 60px;
  }
}

@media (max-width:700px) {
  .c0 {
    height: 50px;
  }
}

@media (max-width:400px) {
  .c0 {
    height: 40px;
  }
}

@media (max-width:900px) {
  .c10 {
    height: 40px;
  }
}

@media (max-width:700px) {
  .c10 {
    height: 30px;
  }
}

@media (max-width:400px) {
  .c10 {
    height: 20px;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <a
        href="/"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="c2"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-home fa-w-18 fa-1x "
            data-icon="home"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </a>
      <span
        className="c3"
      >
        <label
          className="c4"
        >
          <input
            className="c5 c6"
            type="checkbox"
          />
          <span
            className="c7 c8"
          />
        </label>
      </span>
    </div>
    <div
      className="c9"
    >
      <div
        className="c10"
      >
        <input
          className="c11"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
        />
        <div
          className="c12"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 fa-1x "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="c13"
    >
      <div
        className="c2"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user fa-w-14 fa-1x "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>,
  .c0 {
  display: none;
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: rgb(0,0,0);
  background-color: rgba(0,0,0,0.7);
  display: none;
}

.c1 {
  background-color: #fefefe;
  margin: 15% auto;
  padding: 20px;
  border: 1px solid #888;
  border-radius: 5%;
  width: 300px;
  height: 350px;
  position: relative;
}

.c2 {
  color: #aaa;
  position: absolute;
  top: 0px;
  right: 15px;
  font-size: 28px;
  font-weight: bold;
}

.c2:hover,
.c2:focus {
  color: black;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c4 {
  width: 100%;
  height: 30px;
  border: 1px solid #666666;
  margin: 10px 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 20px 0;
}

@media (max-width:900px) {

}

@media (max-width:700px) {

}

@media (max-width:400px) {

}

@media (max-width:900px) {

}

@media (max-width:700px) {

}

@media (max-width:400px) {

}

<div
    className="c0"
    open={false}
  >
    <div
      className="c1"
    >
      <span
        className="c2"
        onClick={[Function]}
      >
        Ã—
      </span>
      <form
        className="c3"
      >
        <input
          className="c4"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          value=""
        />
        <input
          className="c4"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
        <button
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
      </form>
    </div>
  </div>,
]
`;
